source/cloud/aws/ec2.md

```diff
@@ -19,10 +24,113 @@ NVIDIA maintains an [Amazon Machine Image (AMI) that pre-installs NVIDIA drivers
    port `22` and also allow ports `8888,8786,8787` to access Jupyter and Dask. For outbound rules, allow all traffic.
 1. Select **Launch**.
 
+````
+
+````{tab-item} via AWS CLI
+:sync: cli
+
+1. Set the following environment variables first. Edit any of them to match your preferred region, instance type, or naming convention.
+
+   ```bash
+   REGION=us-east-1
+   INSTANCE_TYPE=g5.xlarge
+   KEY_NAME=rapids-ec2-key
+   SG_NAME=rapids-ec2-sg
+   VM_NAME=rapids-ec2
+   ```
+
+2. Accept the NVIDIA Marketplace subscription before using the AMI: open the [NVIDIA GPU-Optimized AMI listing](https://aws.amazon.com/marketplace/pp/prodview-7ikjtg3um26wq), choose **Continue to Subscribe**, then select **Accept Terms**. Wait for the status to show as active.
+
+3. Find the most recent NVIDIA Marketplace AMI ID in `us-east-1`.
+
+   ```bash
+   AMI_ID=$(aws ec2 describe-images \
+       --region "$REGION" \
+       --filters "Name=name,Values=*NVIDIA*VMI*Base*" "Name=state,Values=available" \
+       --query 'Images | sort_by(@, &CreationDate)[-1].ImageId' \
+       --output text)
+   echo "$AMI_ID"
+   ```
+
+4. Create an SSH key pair and secure it locally (if you already have a key, update `KEY_NAME` and skip this step).
+
+   ```bash
+   aws ec2 create-key-pair --region "$REGION" --key-name "$KEY_NAME" \
+       --query 'KeyMaterial' --output text > "${KEY_NAME}.pem"
+   chmod 400 "${KEY_NAME}.pem"
+   ```
+
+5. Create a security group that allows SSH on `22` plus the Jupyter (`8888`) and Dask (`8786`, `8787`) ports, and keep outbound traffic open. Replace `ALLOWED_CIDR` with something more restrictive if you want to limit inbound access. Use `ALLOWED_CIDR="$(curl ifconfig.co)/0"` to restrict access to your current IP address
+
+   ```bash
+   ALLOWED_CIDR=0.0.0.0/0
+   ```
+
+   ```bash
+   VPC_ID=$(aws ec2 describe-vpcs \
+       --region "$REGION" \
+       --query 'Vpcs[0].VpcId' \
+       --output text)
+   echo "$VPC_ID"
```

Comment:
This step isn't choosing my default VPC, it's choosing a project specific one. Which might be why I'm having connection issues.